csharp/src/Google.Protobuf.Test/JsonFormatterTest.cs

```diff
@@ -92,64 +101,146 @@ public void AllSingleFields()
+            var expectedSingleFixed32Value = "23";
+            var expectedSingleFixed64Value = "'1234567890123'";
+            var expectedSingleSfixed32Value = "-123";
+            var expectedSingleSfixed64Value = "'-12345678901234'";
+            var expectedSingleFloatValue = "12.25";
+            var expectedSingleDoubleValue = "23.5";
+            var expectedSingleBoolValue = "true";
+            var expectedSingleStringValue = "'test\\twith\\ttabs'";
+            var expectedSingleBytesValue = "'AQIDBA=='";
+            var expectedSingleNestedMessageValue = "{ 'bb': 35 }";
+            var expectedSingleForeignMessageValue = "{ 'c': 10 }";
+            var expectedSingleImportMessageValue = "{ 'd': 20 }";
+            var expectedSingleNestedEnumValue = "'FOO'";
+            var expectedSingleForeignEnumValue = "'FOREIGN_BAR'";
+            var expectedSingleImportEnumValue = "'IMPORT_BAZ'";
+            var expectedSinglePublicImportMessageValue = "{ 'e': 54 }";
+            var expectedJson = "{ " +
+                "'singleInt32': " + expectedSingleInt32Value + ", " +
+                "'singleInt64': " + expectedSingleInt64Value + ", " +
+                "'singleUint32': " + expectedSingleUint32Value + ", " +
+                "'singleUint64': " + expectedSingleUint64Value + ", " +
+                "'singleSint32': " + expectedSingleSint32Value + ", " +
+                "'singleSint64': " + expectedSingleSint64Value + ", " +
+                "'singleFixed32': " + expectedSingleFixed32Value + ", " +
+                "'singleFixed64': " + expectedSingleFixed64Value + ", " +
+                "'singleSfixed32': " + expectedSingleSfixed32Value + ", " +
+                "'singleSfixed64': " + expectedSingleSfixed64Value + ", " +
+                "'singleFloat': " + expectedSingleFloatValue + ", " +
+                "'singleDouble': " + expectedSingleDoubleValue + ", " +
+                "'singleBool': " + expectedSingleBoolValue + ", " +
+                "'singleString': " + expectedSingleStringValue + ", " +
+                "'singleBytes': " + expectedSingleBytesValue + ", " +
+                "'singleNestedMessage': " + expectedSingleNestedMessageValue + ", " +
+                "'singleForeignMessage': " + expectedSingleForeignMessageValue + ", " +
+                "'singleImportMessage': " + expectedSingleImportMessageValue + ", " +
+                "'singleNestedEnum': " + expectedSingleNestedEnumValue + ", " +
+                "'singleForeignEnum': " + expectedSingleForeignEnumValue + ", " +
+                "'singleImportEnum': " + expectedSingleImportEnumValue + ", " +
+                "'singlePublicImportMessage': " + expectedSinglePublicImportMessageValue +
                 " }";
-            AssertJson(expectedText, actualText);
+            AssertJson(expectedJson, JsonFormatter.Default.Format(message));
+
+            AssertJson(expectedSingleInt32Value, JsonFormatter.Default.Format(message,
```

Comment:
Extract local variables for both the formatter and the TestAllTypes descriptor? I suspect that'll reduce the clutter a certain amount.